acra-censor/handlers/blacklist_handler.go

```diff
@@ -71,6 +82,69 @@ func (handler *BlacklistHandler) CheckQuery(query string) (bool, error) {
 	return true, nil
 }
 
+func (handler *BlacklistHandler) handleAliasedTables(statement *sqlparser.AliasedTableExpr, forbiddenTable string) error {
+	if strings.EqualFold(sqlparser.String(statement.Expr), forbiddenTable) {
+		return ErrAccessToForbiddenTableBlacklist
+	} else {
+		return nil
+	}
+}
+
+func (handler *BlacklistHandler) handleJoinedTables(statement *sqlparser.JoinTableExpr, forbiddenTable string) error {
+	var err error
+	switch statement.LeftExpr.(type) {
+	case *sqlparser.AliasedTableExpr:
+		err = handler.handleAliasedTables(statement.LeftExpr.(*sqlparser.AliasedTableExpr), forbiddenTable)
+	case *sqlparser.JoinTableExpr:
+		err = handler.handleJoinedTables(statement.LeftExpr.(*sqlparser.JoinTableExpr), forbiddenTable)
+	case *sqlparser.ParenTableExpr:
+		err = handler.handleParenTables(statement.LeftExpr.(*sqlparser.ParenTableExpr), forbiddenTable)
+	default:
+		return ErrUnexpectedTypeError
+	}
+
+	if err != nil {
+		return err
+	}
+
+	switch statement.RightExpr.(type) {
+	case *sqlparser.AliasedTableExpr:
+		err = handler.handleAliasedTables(statement.RightExpr.(*sqlparser.AliasedTableExpr), forbiddenTable)
+	case *sqlparser.JoinTableExpr:
+		err = handler.handleJoinedTables(statement.RightExpr.(*sqlparser.JoinTableExpr), forbiddenTable)
+	case *sqlparser.ParenTableExpr:
+		err = handler.handleParenTables(statement.RightExpr.(*sqlparser.ParenTableExpr), forbiddenTable)
+	default:
+		err = ErrUnexpectedTypeError
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (handler *BlacklistHandler) handleParenTables(statement *sqlparser.ParenTableExpr, forbiddenTable string) error {
```

Comment:
handleParenTables -> handleParen**t**Tables?

Reply:
No, it's 'handleParenTables' - like it is used in '*sqlparser.ParenTableExpr'